src/frontend/src/lib/components/transactions/TransactionModal.svelte

```diff
@@ -0,0 +1,125 @@
+<script lang="ts">
+	import { Modal } from '@dfinity/gix-components';
+	import { nonNullish } from '@dfinity/utils';
+	import type { BigNumber } from '@ethersproject/bignumber';
+	import ContentWithToolbar from '$lib/components/ui/ContentWithToolbar.svelte';
+	import Copy from '$lib/components/ui/Copy.svelte';
+	import ExternalLink from '$lib/components/ui/ExternalLink.svelte';
+	import Value from '$lib/components/ui/Value.svelte';
+	import { tokenWithFallback } from '$lib/derived/token.derived';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { modalStore } from '$lib/stores/modal.store';
+	import {
+		formatSecondsToDate,
+		formatToken,
+		shortenWithMiddleEllipsis
+	} from '$lib/utils/format.utils';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	export let from: string;
```

Comment:
I extracted common fields into a separate type (`TransactionUiCommon`, can be used by other networks) and passing them as an object to TransactionModal. There are still some props that passed outside of that object because most likely those values will changed to `slots` when adjusting `TransactionModal` for usage by ETH and IC.

Reply:
Thanks a lot. It's not exactly what I had in mind, but it looks good too. However, after comparing both the old and new versions, I'm actually not sure anymore which one is better. Which solution do you prefer?

Please pick the one you rather like.

Reply:
I will keep as it is for now since there is still work to do with this component, so in the end I hope it will look better than it is now 🙂